kgforge/specializations/models/rdf/service.py

```diff
@@ -162,18 +181,20 @@ def validate(self, resource: Resource, type_: str):
                     "Resource has list of types as attribute and type_ parameter is not specified. "
                     "Please provide a type_ parameter to validate against it."
                 )
-            if type_ is None:
-                shape_iri = self.types_to_shapes[resource.type]
-            else:
-                shape_iri = self.types_to_shapes[type_]
+            type_to_validate = type_ if type_ else resource.type
```

Comment:
1. I would expect that even if a resource does not have a type attribute, one could validate if a type_ was provided. This is not the case because we check whether resource.type is a list, and that would always raise an AttributeError, even if type_ was not None. So first type_ is None should be checked, then in a nested if, resource.type being a list should be checked. 

2. shape_iri = self.get_ ... should not be within the try block

Reply:
> I would expect that even if a resource does not have a type attribute, one could validate if a type_ was provided. 

The current PR is doing that. If type_ is provided then it will be used if not resource.type is used unless it is a list.